File: backend/ESESMT.Domain/Entities/CompletedChecklist.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ESESMT.Domain.Entities
{
    public class CompletedChecklist : BaseEntity<int>
    {
        public DateTime CreationDate { get; set; }
        public int ChecklistId { get; set; }

        public Checklist Checklist { get; set; }
        public ICollection<CompletedChecklistItem> CompletedChecklistItems { get; set; }
    }
}

File: frontend/src/app/shared/models/completed-checklist.ts
import { CompletedChecklistItem } from "./completed-checklist-item";

export class CompletedChecklist {
    id: number;
    checklistId: number;
    checklistDescription: string;
    checklistChecklistTypeName: string;
    creationDate: string;
    completedChecklistItems: Array<CompletedChecklistItem>;
}